Releases - DevOps for ASP.NET Core Developers eBook V1.1 (#23784)
dotnet/docs: docs/architecture/devops-for-aspnet-developers/cicd.md

1. Select the **GitHub** option from the **Select a source** section:

    ![Select a source - GitHub](media/cicd/vsts-select-source.png)

1. Authorization is required before Azure DevOps can access your GitHub repository. Enter *<GitHub_username> GitHub connection* in the **Connection name** textbox. For example:

    ![GitHub connection name](media/cicd/vsts-repo-authz.png)

1. If two-factor authentication is enabled on your GitHub account, a personal access token is required. In that case, click the **Authorize with a GitHub personal access token** link. See the [official GitHub personal access token creation instructions](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) for help. Only the *repo* scope of permissions is needed. Otherwise, click the **Authorize using OAuth** button.
1. When prompted, sign in to your GitHub account. Then select Authorize to grant access to your Azure DevOps organization. If successful, a new service endpoint is created.
1. Click the ellipsis button next to the **Repository** button. Select the *<GitHub_username>/simple-feed-reader* repository from the list. Click the **Select** button.

> 1. Select the default branch (*master*) from the **Default branch for manual and scheduled builds** drop-down. Click the **Continue** button. The template selection page appears.

Reviewer: main?

Author: Yes, modified. Thank you.

1. Click **Deployment Center**. A new panel appears. Click **Disconnect** to remove the local Git source control configuration that was added in the previous chapter. Confirm the removal operation by clicking the **Yes** button.
1. Navigate to the *mywebapp<unique_number>* App Service. As a reminder, the portal's search box can be used to quickly locate the App Service.
1. Click **Deployment Center**. A new panel appears. Click **Disconnect** to remove the local Git source control configuration that was added in the previous chapter. Confirm the removal operation by clicking the **Yes** button.

## Create an Azure DevOps organization

1. Open a browser, and navigate to the [Azure DevOps organization creation page](https://go.microsoft.com/fwlink/?LinkId=307137).
1. Select **New organization**
1. Confirm the information, and then select **Continue**.
1. Sign in to your organization at any time, `https://dev.azure.com/{yourorganization}`

## Create a team project in Azure DevOps organization

1. Choose the organization, and then select **New project**.
1. Enter the project name as *MyFirstProject* and select the **Visibility** as *Private*
1. Select **Create project**.

> For more information, see the [Create a project](azure/devops/organizations/projects/create-project?view=azure-devops&tabs=preview-page#create-a-project)

Reviewer 2: ```suggestion
For more information, see [Create a project](/azure/devops/organizations/projects/create-project?view=azure-devops&tabs=preview-page&preserve-view=true#create-a-project)
```

## Configure a self-hosted agent if necessary

> To build your code or deploy your software using Azure Pipelines, you need at least one agent. In Azure Pipelines, you can run parallel jobs on either **Microsoft-hosted** or **self-hosted** agent. But with the recent change in Azure Pipelines free grant of parallel jobs is temporarily disable for the public projects.For more details, refer [Configure and pay for parallel jobs](azure/devops/pipelines/licensing/concurrent-jobs?view=azure-devops&tabs=ms-hosted).

Reviewer 2: ```suggestion
To build your code or deploy your software using Azure Pipelines, you need at least one agent. In Azure Pipelines, you can run parallel jobs on either **Microsoft-hosted** or **self-hosted** agent. But with the recent change in Azure Pipelines free grant of parallel jobs is temporarily disable for the public projects.For more details, refer [Configure and pay for parallel jobs](/azure/devops/pipelines/licensing/concurrent-jobs?view=azure-devops&tabs=ms-hosted&preserve-view=true).
```

Go to **Organization Settings** and then **Pipelines** > **Parallel jobs**. If you see value **0** under **Microsoft-hosted** that means you need a **Self-hosted** agent to run your pipeline.

![MS-hosted agent](media/cicd/azure-devops-ms-hosted-agent.png)

> You can create that by following details mentioned in [Self-hosted agents](azure/devops/pipelines/agents/agents?view=azure-devops&tabs=browser#install). After successful configuration, you'll be able to see available agent under **Organization Settings** > **Agent pools** > **{youragentname}**

Reviewer 2: ```suggestion
You can create that by following details mentioned in [Self-hosted agents](/azure/devops/pipelines/agents/agents?view=azure-devops&tabs=browser&preserve-view=true#install). After successful configuration, you'll be able to see available agent under **Organization Settings** > **Agent pools** > **{youragentname}**
```
